### PR TITLE
Teamview rework

### PR DESCRIPTION
**Pull request by kbats183**

### CODE REVIEW

File: src/frontend/overlay/src/components/molecules/info/ContestantViewCorner.tsx [hunk 1]
```diff
@@ -9,25 +9,43 @@ import { useAppSelector } from "@/redux/hooks";
 
 const ContestantViewCornerWrap = styled.div<{isSmall: boolean}>`
   display: grid;
-  grid-auto-rows: ${c.QUEUE_ROW_HEIGHT}px;
-  grid-template-columns: auto minmax(100px, 150px);
+  grid-auto-rows: 1fr ${c.QUEUE_ROW_HEIGHT}px;
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/molecules/info/ContestantViewCorner.tsx:12]: https://css-tricks.com/almanac/properties/g/grid-auto-rows/#aa-multiple-track-sizes

File: src/frontend/overlay/src/components/molecules/info/ContestantViewCorner.tsx [hunk 2]
```diff
@@ -9,25 +9,43 @@ import { useAppSelector } from "@/redux/hooks";
...
+    grid-column-start: 1;
+    grid-column-end: 3;
+    grid-row-start: 1;
+    grid-row-end: 2;
+
+    display: flex;
+    flex-wrap: wrap;
+    flex-direction: column;
+    align-content: flex-start;
+    /* direction: rtl; */
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/molecules/info/ContestantViewCorner.tsx:38]: what?

File: src/frontend/overlay/src/components/molecules/info/ContestantViewCorner.tsx [hunk 3]
```diff
@@ -9,25 +9,43 @@ import { useAppSelector } from "@/redux/hooks";
...
+    transform: scale(-1);
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/molecules/info/ContestantViewCorner.tsx:39]: What?

File: src/frontend/overlay/src/components/molecules/info/ContestantViewCorner.tsx [hunk 4]
```diff
@@ -9,25 +9,43 @@ import { useAppSelector } from "@/redux/hooks";
...
+    overflow: hidden;
+    height: 100%;
+`;
+
+const TaskRow = styled.div`
+  display: flex;
+  width: 150px;
+  flex: 0 0 ${c.QUEUE_ROW_HEIGHT - 1}px;
+  transform: scale(-1);
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/molecules/info/ContestantViewCorner.tsx:48]: what?

File: src/frontend/overlay/src/components/molecules/info/ContestantViewLine.tsx [hunk 5]
```diff
@@ -49,16 +49,19 @@ const TaskRow = styled.div<TaskRowProps>`
 
   width: 100%;
 
-  border-radius: ${props => props.index === 0 && props.roundTop ? "16px" : 0} ${props => props.index === props.problems - 1 && props.roundTop ? "16px" : 0}  ${props => props.index === props.problems - 1 && props.roundBottom ? "16px" : 0} ${props => props.index === 0 && props.roundBottom ? "16px" : 0};
+  border-radius: ${props => props.index === 0 && props.roundTop ? c.GLOBAL_BORDER_RADIUS : 0} 
+      ${props => props.index === props.problems - 1 && props.roundTop ? c.GLOBAL_BORDER_RADIUS : 0}
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/molecules/info/ContestantViewLine.tsx:53]: Maybe don't pass problems to props, and instead pass isFirstRow isLastRow?  
Or better use :first-child :last-child pseudoselectors

File: src/frontend/overlay/src/components/organisms/holder/TeamMediaHolder.tsx [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+
+type MediaHolderProps<M extends MediaType> = {
+    media: M;
+    onLoadStatus: (currentStatus: boolean) => void;
+    className?: string;
+};
+
+const useTeamMediaAudio = () => {
+    const queryParams = useQueryParams();
+    return !queryParams.has("teamMediaAudio") && window["obsstudio"] ||
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/organisms/holder/TeamMediaHolder.tsx:18]: Add brackets for clarity

File: src/frontend/overlay/src/components/organisms/holder/TeamMediaHolder.tsx [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+import mpegts from "mpegts.js";
+import { GrabberPlayerClient } from "../../../utils/grabber/grabber_player";
+
+type MediaHolderProps<M extends MediaType> = {
+    media: M;
+    onLoadStatus: (currentStatus: boolean) => void;
+    className?: string;
+};
+
+const useTeamMediaAudio = () => {
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/organisms/holder/TeamMediaHolder.tsx:16]: `useIsTeamMediaAudio`

File: src/frontend/overlay/src/components/organisms/widgets/TeamView.tsx [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+    display: ${props => props.show ? "flex" : "none"};
+    flex-direction: column;
+    justify-content: start;
+    align-items: flex-end;
+    position: relative;
+    animation: ${props => props.animation} ${c.TEAM_VIEW_APPEAR_TIME}ms ${props => props.animationStyle};
+    animation-fill-mode: forwards;
+`;
+
+type VariantProps = {
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/organisms/widgets/TeamView.tsx:41]: ?CommonContentProps?

**Review by irdkwmnsb — APPROVED** [review]
Nicely done!

File: src/cds/core/src/main/kotlin/org/icpclive/cds/api/MediaType.kt [hunk 9]
```diff
@@ -9,23 +9,23 @@ public sealed class MediaType {
 
     @Serializable
     @SerialName("Image")
-    public data class Image(val url: String, override val isMedia: Boolean = true) : MediaType()
+    public data class Image(val url: String, override val isMedia: Boolean = true, val vertical: Boolean = false) : MediaType()
```
Comment (kunyavskiy) [inline src/cds/core/src/main/kotlin/org/icpclive/cds/api/MediaType.kt]: Maybe you want to make it `@Requied` to always have value on frontend

**Review by kunyavskiy — APPROVED** [review]

**Pull request merged**